dvc/repo/get.py

```diff
@@ -43,7 +49,8 @@ def get(url, path, out=None, rev=None):
     dpath = os.path.dirname(os.path.abspath(out))
     tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
     try:
-        with external_repo(cache_dir=tmp_dir, url=url, rev=rev) as repo:
+        cached_clone(url, rev=rev, clone_path=tmp_dir)
```

Comment:
You don't need to pass tmp_dir to cached_clone. We do that in `external_repo` because we want our cache to be located right beside the location that we want to checkout it to to speed it up and have an ability to use reflinks and hardlinks, instead of copying from another filesystem. Btw, you've lost that optimization in your change.

Reply:
I thought this was enough, since the Repo is rooted at `tmp_dir` I suppose it's being cloned right there, and then copied from the same FS? What am I missing?

Reply:
That is a good point! I've missed that! Indeed, when doing so you no longer need to change cache dir. The reason we do that is that there is a chance that the fs that we are currently on doesn't have proper support for some stuff like locking (e.g. https://www.sqlite.org/howtocorrupt.html#_filesystems_with_broken_or_missing_lock_implementations), which might make `Repo()` freeze. So it is best to keep it the way it was before.